projects/js-packages/publicize-components/src/components/form/index.js

```diff
@@ -125,15 +134,20 @@ export default function PublicizeForm( {
 		[ dismissNotice, NOTICES ]
 	);
 
-	const pageHasFocus = usePageHasFocus();
+	const shouldAutoRefresh = useRef( false );
+	const enableAutoRefresh = useCallback( () => {
+		shouldAutoRefresh.current = true;
+	}, [ shouldAutoRefresh ] );
+
+	const pageHasFocus = usePageVisibility();
 	const refreshOptions = useDispatch( SOCIAL_STORE_ID ).refreshAutoConversionSettings;
 
 	const isAutoConversionEnabled = useSelect(
 		select => select( SOCIAL_STORE_ID ).isAutoConversionEnabled(),
 		[]
 	);
 
-	if ( pageHasFocus ) {
+	if ( pageHasFocus && shouldAutoRefresh.current ) {
```

Comment:
We ended up debouncing the requests for connection-test-results in case someone is flicking back and forth between tabs. Maybe we should do the same here?

I'm not sure why we debounced rather than throttled, but we might want to use throttle instead so an initial call is made quickly but we don't make any further calls for a few seconds.